### PR TITLE
Add a new derive 'perf_with_debug'

### PR DESCRIPTION
**Pull request by pmnoxx**
Add a new drive perf_with_debug, this should allow printing the type of message inside near-performance-metrics::measure_performance

### CODE REVIEW

**Comment by pmnoxx** [comment]
This is a duplicate of #3766

File: utils/near-performance-metrics-macros/src/lib.rs [hunk 1]
```diff
@@ -35,3 +35,35 @@ pub fn perf(_attr: TokenStream, item: TokenStream) -> TokenStream {
         panic!("not a function {:?}");
     }
 }
+
+#[proc_macro_attribute]
+pub fn perf_with_debug(_attr: TokenStream, item: TokenStream) -> TokenStream {
```
Comment (bowenwang1996) [inline utils/near-performance-metrics-macros/src/lib.rs:40]: there seems to be quite some code duplication with `perf`

**Review by bowenwang1996 — COMMENTED** [review]
Can we add some comments documenting the procedural macros `perf` and `perf_with_debug`?

**Comment by bowenwang1996** [comment]
@pmnoxx is this ready for review?

**Comment by pmnoxx** [comment]
> @pmnoxx is this ready for review?

@bowenwang1996 Yes, the change is ready for review.

**Review by bowenwang1996 — APPROVED** [review]
Would be nice if you can address https://github.com/near/nearcore/pull/3772#pullrequestreview-561423216

**Comment by pmnoxx** [comment]
> Would be nice if you can address [#3772 (review)](https://github.com/near/nearcore/pull/3772#pullrequestreview-561423216)

Oh, sorry I missed your message. I just added the documentation.
I'm thinking about changing `perf_with_debug` to maybe `perf_with_asstaticstr`. In case we decide to add to variants of `perf` in the future. What do you think?

**Pull request merged**